src/go/configmanager/config_manager.go

```diff
@@ -154,64 +146,52 @@ func NewConfigManager(mf *metadata.MetadataFetcher, opts options.ConfigGenerator
 				return nil, fmt.Errorf("service config id is not specified, required on a non-gcp deployment")
 			}
 		}
-		m.curConfigID = configID
-		if err := m.updateSnapshot(); err != nil {
-			return nil, err
-		}
+		serviceConfig, err = m.serviceConfigFetcher.FetchConfig(configID)
```

Comment:
Why don't you move `configID` into constructor `NewServiceConfigFetcher`. It's only ever needed once.